src/components/Modals/SelectSemesterDuplicate.vue

```diff
@@ -0,0 +1,305 @@
+          <div
+            class="selectSemester-dropdown-placeholder season-wrapper"
+            @click="showHideSemesterContent"
+            data-cyId="newSemester-seasonWrapper"
+          >
+            <!-- text to display in dropdown -->
+            <div
+              class="selectSemester-dropdown-placeholder season-placeholder"
+              :style="{ color: displayOptions.semester.placeholderColor }"
+            >
+              {{ semesterPlaceholder }}
+            </div>
+            <div
+              class="selectSemester-dropdown-placeholder season-arrow"
+              :style="{ borderTopColor: displayOptions.semester.arrowColor }"
+            ></div>
+          </div>
+          <div
+            class="selectSemester-dropdown-content season-content position-absolute w-100"
+            v-if="displayOptions.semester.shown"
+          >
+            <!-- show the list of semesters course shows up in -->
+            <div
+              v-for="s in semestersTakenList"
+              :key="s"
+              class="selectSemester-dropdown-content-item"
+              @click="selectSemester(s)"
+              data-cyId="newSemester-seasonItem"
+            >
+              {{ s }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { PropType, defineComponent } from 'vue';
+import { clickOutside } from '@/utilities';
+import { inactiveGray, yuxuanBlue, darkPlaceholderGray } from '@/assets/constants/scss-variables';
+
+type DisplayOption = {
+  shown: boolean;
+  stopClose: boolean;
+  boxBorder: string;
+  arrowColor: string;
+  placeholderColor: string;
+};
+
+type Data = {
+  semesterText: string;
+  readonly displayOptions: {
+    readonly semester: DisplayOption;
+  };
+};
+/**
+ * Dropdown bar to resolve duplicates of the same course with the list of
+ * semesters that this course exists in
+ */
+export default defineComponent({
+  props: {
+    isEdit: { type: Boolean, default: false },
+    isCourseModelSelectingSemester: { type: Boolean, default: false },
+    semestersTaken: { type: Array as PropType<readonly FirestoreSemester[]>, required: true },
+  },
+  emits: {
+    updateSemProps: (season: string): boolean => typeof season === 'string',
+    duplicateSemester: (duplicate: boolean): boolean => typeof duplicate === 'boolean',
+  },
+  data(): Data {
+    const placeholderColor = darkPlaceholderGray;
+    return {
+      semesterText: '',
+      displayOptions: {
+        semester: {
+          shown: false,
+          stopClose: false,
+          boxBorder: '',
+          arrowColor: '',
+          placeholderColor,
+        },
+      },
+    };
+  },
+  computed: {
+    // shows the semester taken or 'Select Semester'
+    semesterPlaceholder(): string {
+      return this.semesterText || 'Select Semester';
+    },
+    semestersTakenList(): string[] {
```

Comment:
Can you add some documentation for this function and the motivation for how it's implemented?